src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,49 +1,97 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    public static final int DEFAULT_CAPACITY = 10;
+    private int capacity;
+    private T[] array;
+    private int size;
+
+    public ArrayList() {
+        capacity = DEFAULT_CAPACITY;
+        array = (T[]) new Object[capacity];
+        size = 0;
+    }
 
     @Override
     public void add(T value) {
-
+        grow();
+        array[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
-
+        isIndexCorrect(index, index > size);
+        grow();
+        System.arraycopy(array, index, array, index + 1, size - index);
```

Comment:
```suggestion
        isIndexCorrect(index, index > size);
        System.arraycopy(array, index, array, index + 1, size - index);
```
тут не потрібно так як цей випадок хендлиться під час arraycopy рядком нижче